### PR TITLE
Furtastic: Image upload support on PostyBirb

### PR DESCRIPTION
**Pull request by Xenoyia**
This PR is to enable image upload support on the Furtastic platform, for accounts that have an API key enabled.

I have recreated the PR as it was extremely messy with a ton of confusing commits by me. :(

### CODE REVIEW

**Review by leaftail1880 — APPROVED** [review]

File: commons/src/interfaces/websites/furtastic/furtastic.account.interface.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface furtasticAccountData {
```
Comment (mvdicarlo) [inline commons/src/interfaces/websites/furtastic/furtastic.account.interface.ts:1]: PascalCase

File: commons/src/interfaces/websites/furtastic/furtastic.file.options.interface.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { DefaultFileOptions } from '../../submission/default-options.interface';
+
+// tslint:disable-next-line: class-name
+export interface furtasticFileOptions extends DefaultFileOptions {
```
Comment (mvdicarlo) [inline commons/src/interfaces/websites/furtastic/furtastic.file.options.interface.ts:4]: PascalCase
Remove tslint

File: electron-app/src/server/websites/furtastic/furtastic.service.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  }
+
+  preparseDescription(text: string) {
+    return UsernameParser.replaceText(text, 'ft', '@$1').replace(
+      /<a(.*?)href="(.*?)"(.*?)>(.*?)<\/a>/gi,
+      '"$4":$2',
+    );
+  }
+
+  parseDescription(text: string) {
```
Comment (mvdicarlo) [inline electron-app/src/server/websites/furtastic/furtastic.service.ts:66]: Just to make sure, your website uses bbcode tags?

File: electron-app/src/server/websites/furtastic/furtastic.service.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    }) as any;
+
+    if (post.body.status && post.body.status === true) {
+      return this.createPostResponse({ source: `https://furtastic.art${post.body.url}` });
+    }
+
+    return Promise.reject(
+      this.createPostResponse({
+        additionalInfo: JSON.stringify(post.body),
+        message: ` ${post.body.message}`,
```
Comment (mvdicarlo) [inline electron-app/src/server/websites/furtastic/furtastic.service.ts:129]: Does this need to be wrapped in ``?

**Comment by leaftail1880** [comment]
Reminder about eslint vscode extension and `npm run lint`

**Comment by leaftail1880** [comment]
By the way, there is already markdown parser https://github.com/mvdicarlo/postybirb-plus/blob/develop/electron-app/src/server/websites/furry-network/furry-network.service.ts:52

**Comment by Xenoyia** [comment]
> By the way, there is already markdown parser https://github.com/mvdicarlo/postybirb-plus/blob/develop/electron-app/src/server/websites/furry-network/furry-network.service.ts:52

I will work on cleaning it up and making it more efficient with the parser in an update! It works currently but obviously needs to be cleaned up a little ^^

**Comment by leaftail1880** [comment]
May i update it to use MarkdownParser.parse? I need to do some cleanups too heh

**Comment by Xenoyia** [comment]
> May i update it to use MarkdownParser.parse? I need to do some cleanups too heh

Help yourself! I'd appreciate any help with this :D

**Pull request merged**